apps/api/src/user/routes/register-user/register-user.router.ts

```diff
@@ -0,0 +1,52 @@
+import { createRoute } from "@hono/zod-openapi";
+import { container } from "tsyringe";
+import { z } from "zod";
+
+import { OpenApiHonoHandler } from "@src/core/services/open-api-hono-handler/open-api-hono-handler";
+import { UserController } from "@src/user/controllers/user/user.controller";
+import { UserSchema } from "@src/user/schemas/user.schema";
+
+const registerUserInputSchema = z.object({
+  wantedUsername: z.string(),
+  email: z.string(),
+  emailVerified: z.boolean(),
+  subscribedToNewsletter: z.boolean().optional()
+});
+export type RegisterUserInput = z.infer<typeof registerUserInputSchema>;
+
+const registerUserResponseSchema = z.object({
+  data: UserSchema
+});
+export type RegisterUserResponse = z.infer<typeof registerUserResponseSchema>;
+
+const route = createRoute({
+  method: "post",
+  path: "/v1/registerUser",
```

Comment:
suggestion(non-blocking): we use lower case dash for new routes
```suggestion
  path: "/v1/register-user", // or /v1/register might be simpler
```